network/address_test.go

```diff
@@ -27,6 +29,12 @@ func TestConnType(t *testing.T) {
 }
 
 func TestAddress(t *testing.T) {
+	localhostNetworkAddresses, _ := net.LookupHost("localhost")
+	sort.Strings(localhostNetworkAddresses)
+	facebookNetworkAddresses, _ := net.LookupHost("facebook.com")
+	sort.Strings(facebookNetworkAddresses)
+	googleNetworkAddresses, _ := net.LookupHost("google.com")
```

Comment:
The fact is that such change (the additional parameter, representing how the hostname is resolved) would be needed for all the functions that rely on NetworkAddress() (like Host() or Port()). Does this really make sense?